newm-chain/src/main/kotlin/io/newm/chain/daemon/BlockDaemon.kt

```diff
@@ -615,7 +623,10 @@ class BlockDaemon(
                                                 ix = utxo.ix
                                                 lovelace = utxo.lovelace.toString()
                                                 utxo.datumHash?.let { datumHash = it }
-                                                utxo.datum?.let { datum = it }
+                                                utxo.datum?.let {
+                                                    datum = CborReader.createFromByteArray(it.hexToByteArray())
+                                                        .readDataItem().toPlutusData(it)
```

Comment:
Maybe a local private function can be created for these 2 lines since its used in several places.

Reply:
fixed